pkg/provider/okta/okta.go

```diff
@@ -160,7 +165,26 @@ func (oc *Client) Authenticate(loginDetails *creds.LoginDetails) (string, error)
 
 	samlAssertion, ok := doc.Find("input[name=\"SAMLResponse\"]").Attr("value")
 	if !ok {
-		return samlAssertion, errors.Wrap(err, "unable to locate saml response")
+		req, err = http.NewRequest("GET", loginDetails.URL, nil)
+		if err != nil {
+			return samlAssertion, errors.Wrap(err, "error building app request")
+		}
+		res, err = oc.client.Do(req)
+		if err != nil {
+			return samlAssertion, errors.Wrap(err, "error retrieving app response")
+		}
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return "", errors.Wrap(err, "error retrieving body from response")
+		}
+		resp = string(body)
+		re := regexp.MustCompile("var stateToken = '(.*)';")
```

Comment:
Would be good to lift this section out to a standalone function and write a small test with a couple of examples.

Reply:
See https://github.com/Versent/saml2aws/pull/369/commits/3221a0d5ac95dae02e7fb2284366b223c5f3822b